lib/utils/mailup.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Implements MailUp REST API
+ */
+
+import * as request from "request";
+import * as winston from "winston";
+
+import { Either, isLeft, left, right } from "fp-ts/lib/Either";
+
+// request timeout in milliseconds
+const TIMEOUT = 10000;
+
+type HttpMethod = "GET" | "POST";
+
+export interface ISmtpAuthInfo {
+  readonly Username: string;
+  readonly Secret: string;
+}
+
+export interface IApiResponse {
+  readonly Status: string;
+  readonly Code: string;
+  readonly Message: string;
+}
+
+function callMailUpApi<T extends IApiResponse>(
+  method: HttpMethod,
+  url: string,
+  creds: ISmtpAuthInfo,
+  payload: {}
+): Promise<Either<Error, T>> {
+  return new Promise(resolve => {
+    const options = {
+      uri: url,
+      method,
+      json: { ...payload, User: creds },
+      timeout: TIMEOUT
+    };
+    request(options, async (err, res, body) => {
+      if (err) {
+        winston.error("callMailUpApiInternal|error|" + JSON.stringify(err));
+        return resolve(left<Error, T>(new Error(err)));
+      }
+      if (res.statusCode === 200) {
+        return resolve(right<Error, T>(body));
+      }
+      winston.error("callMailUpApiInternal|error|" + JSON.stringify(res));
+      return resolve(left<Error, T>(new Error("Cannot get access token")));
+    });
+  });
+}
+
+export async function sendTransactionalMail<T extends IApiResponse>(
+  creds: ISmtpAuthInfo,
+  payload: {}
```

Comment:
payload is an arbitrary object?

Reply:
no, adding a type for it is on my todo list